test/server/listener_manager_impl_test.cc

```diff
@@ -749,5 +750,237 @@ TEST_F(ListenerManagerImplTest, EarlyShutdown) {
   manager_->stopWorkers();
 }
 
+TEST_F(ListenerManagerImplWithRealFiltersTest, SniWithSingleFilterChain) {
+  Server::Configuration::HttpConnectionManagerFilterConfigFactory factory;
```

Comment:
Why are those factory local variables needed? AFAICT they are not used?

Reply:
In order to register `envoy.http_connection_manager`, otherwise loading of the configuration will fail because of an unknown filter.

Reply:
I see. I think the issue is not this variable, but just linking against the translation unit that has the config factory in it. (Since that is where static registration happens). I don't think this variable actually does anything. If you leave the dep and remove the variable and include does it pass?

Reply:
Good catch, linking against it is enough to fix that. Thanks!